src/lwc/rd2EntryForm/rd2EntryForm.js

```diff
@@ -381,35 +431,196 @@ export default class rd2EntryForm extends LightningElement {
-            : insertSuccessMessage.replace("{0}", recordName);
+        this.recordName = event.detail.fields.Name.value;
+        const recordId = event.detail.id;
+
+        if (this.isElevateWidgetEnabled && this.paymentMethodToken) {
+            this.handleElevateCommitment(recordId);
+
+        } else {
+            this.showToastSuccess();
+            this.closeModal(recordId);
+        }
+    }
+
+    /**
+     * @description Sends Commitment request to the Elevate API and
+     * updates the Recurring Donation Commitment Id on the successfull creation
+     * If an error occurs when the Commitment is created,
+     * it should be displayed, and the modal closed since
+     * the another creation of the same Recurring Donation should be prevented.
+     */
+    handleElevateCommitment(recordId) {
+        this.loadingText = this.customLabels.creatingCommitmentMessage;
+        
+        getCommitmentRequestBody({
+            recordId: recordId,
+            paymentMethodToken: this.paymentMethodToken
+        })
+            .then(jsonRequestBody => {
+                createCommitment({ record: recordId, jsonRequestBody: jsonRequestBody })
+                    .then(jsonResponse => {
+                        this.processCommitmentResponse(jsonResponse);
+                    })
+                    .catch((error) => {
+                        this.displayCommitmentError(error);
+                    })
+                    .finally(() => {
+                        this.closeModal(recordId);
+                    });
+            })
+            .catch(error => {
+                this.displayCommitmentError(error);
+                this.closeModal(recordId);
+            });        
+    }
+
+    /**
+     * @description Verifies if the Commitment was successfully created,
+     * and displays the error if the error response has been returned.
+     */
+    processCommitmentResponse(jsonResponse) {
+        let response = JSON.parse(jsonResponse);
+
+        if (response.statusCode === HTTP_CODES.Created) {
+            this.showToastSuccess();
 
-        showToast(message, '', 'success', []);
+        } else {
+            this.displayCommitmentErrorResponse(response);
+        } 
+    }
+
+    /**
+     * @description Displayes errors extracted from the error response
+     * returned from the Elevate API when the Commitment cannot be created
+     */
+    displayCommitmentErrorResponse(response) {
+        if (response.body) {
+            // Errors returned in the response body can contain a single quote, for example:
+            // "body": "{'errors':[{'message':'\"Unauthorized\"'}]}".
+            // However, the JSON parser does not work with a single quote, 
+            // so need to replace it with the double quote but after
+            // replacing \" with an empty string, otherwise the message content
+            // will be misformatted.
+            if (JSON.stringify(response.body).includes("'errors'")) {
```

Comment:
Is this the json structure work-around you mentioned in slack?

Reply:
Yes, this workaround is required so the error response containing `"body": "{'errors':[{'message':'\"Unauthorized\"'}]}" ` can be parsed and processed.